### PR TITLE
fix for cancellation honoring and compat issue

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Honors cancellation of publishing test results when build is cancelled.
Honors compat with old tasks(testresultfiles gives as data instead of testResuls parameter.).
Testing : manual , unit test.

PS : Run is not aborted when the build is cancelled.

### CODE REVIEW

File: src/Agent.Worker/TestResults/ResultsCommands.cs [hunk 1]
```diff
@@ -18,28 +20,14 @@ public class ResultsCommands : AgentService, ICommandExtension
         private string _configuration;
         private string _runTitle;
         private bool _publishRunLevelAttachments;
+        private int runCounter = 0;
```
Comment (TingluoHuang) [inline src/Agent.Worker/TestResults/ResultsCommands.cs]: _runCounter

File: src/Agent.Worker/TestResults/ResultsCommands.cs [hunk 2]
```diff
@@ -18,28 +20,14 @@ public class ResultsCommands : AgentService, ICommandExtension
...
+        object sync = new object();
```
Comment (TingluoHuang) [inline src/Agent.Worker/TestResults/ResultsCommands.cs]: readonly _sync

File: src/Agent.Worker/TestResults/ResultsCommands.cs (hunk 1 above)
Comment (AshwiniChalla) [inline src/Agent.Worker/TestResults/ResultsCommands.cs]: missed it.

**Comment by bryanmacfarlane** [comment]
We have an update going out very soon.  FYI in the event this should be in ...

**Comment by acesiddhu** [comment]
looks fine

**Pull request merged**